scratch/app-github.jsonc

```diff
@@ -0,0 +1,39 @@
+{
+    "type": "view",
+    "bindings": [
+        {
+            "type": "menu",
+            "bindings": [
+                // Are these menu items meant to lead to rendering the list of open PRs with Mattermost, or is it a link to a page on GitHub with these listed?
```

Comment:
A question for design. There are currently no designs for listing GitHub PRs in the RHS, but I imagine it would be similar to what the GitHub plugin is currently doing for this feature, though it would be less detailed in this case.

Reply:
@mickmister this screen shows PRs in a listing in the RHS...and yes, similar to the current Github plugin - just cleaned up a bit: https://www.figma.com/file/EGtWDFzovP6NznvVARhETM/MM-34060-Global-Layout-Updates-v2?node-id=396%3A172665

Reply:
@matthewbirtch My bad for missing this. I was going off of the designs I had already collected here for the effort of designing the data model https://docs.google.com/document/d/1_v6eDswUlil3O3gVMo0_YQXCyquggvJ9Jm_zqop1y_E/edit.

I think there's an opportunity to support a "Chip Group" (possibly different name) and nested Chips to implement the design you've linked. It would need to fit the data model [here](https://pkg.go.dev/github.com/mattermost/mattermost-plugin-apps/apps#Binding) or have the data model be extended to fit this. At the moment, we don't have the concept of "color", but we can add this or repurpose an existing field to support it, such as the "hint" field.

Using a field like "hint" for color, that seems like it has nothing to do with color, is taking inspiration from how [Reddit's database](http://highscalability.com/blog/2010/5/17/7-lessons-learned-while-building-reddit-to-270-million-page.html) is designed:

> Everything in Reddit is a Thing: users, links, comments, subreddits, awards, etc. Things keep common attribute like up/down votes, a type, and creation date. The Data table has three columns: thing id, key, value. There’s a row for every attribute. There’s a row for title, url, author, spam votes, etc. When they add new features they didn’t have to worry about the database anymore. They didn’t have to add new tables for new things or worry about upgrades. Easier for development, deployment, maintenance.

The names of the text fields in our data model (label, description, hint) are arbitrary in the grand scheme of things. They are reusable and can mean different things in different contexts. I see a collection of Chips potentially being defined as:

```json
{
    "type": "chip_group",
    "bindings": [
        {
            "label": "1: UX Review",
            "hint": "#7CDFE2"
        },
        {
            "label": "2: Dev Review",
            "hint": "#EB6420"
        },
        {
            "label": "3: QA Review",
            "hint": "#7CDFE2"
        }
    ]
}
```